heron/scheduler-core/src/java/com/twitter/heron/scheduler/SubmitterMain.java

```diff
@@ -425,6 +428,10 @@ public void submitTopology() throws TopologySubmissionException {
       // Check if topology is already running
       validateSubmit(adaptor, topology.getName());
 
+      // Try to clear left over data in case the previous kill operation
```

Comment:
How do you distinguish the case between 
- Normal running topology
- Topology in a partial state

Reply:
if a topology has defn and executionState data, it is considered as "running", otherwise it is considered as "not running". This is the validation logic in RuntimeManager, and i updated the validateSubmit() called right above to follow the same condition in this PR.

I was thinking of using defn data as the source of truth, but executionState is very important in RuntimeManager validation. So here I am keep both. Another option is to use executionState only as source of truth, if you think it is better that way.

When the topology is "not running", it should be safe to call cleanState().